fractal_tasks_core/cellpose_segmentation.py

```diff
@@ -55,7 +58,223 @@
+        raise ValueError(
+            f"Wrong arguments for preprocess_cellpose_input:\n"
+            f"{use_masks=}\n"
+            f"{current_label_path=}\n, "
+            f"{ROI_index=}\n"
+            f"{region=}\n"
+            f"{ROI_table_path=}"
+        )
+
+    # Load the ROI table and its metadata attributes
+    ROI_table = ad.read_zarr(ROI_table_path)
+    ROI_table_obs = ROI_table.obs
+    attrs = zarr.group(ROI_table_path).attrs
+    if not attrs["type"] == "ngff:region_table":
+        raise ValueError("Wrong attributes for {ROI_table_path}:\n{attrs}")
+    label_relative_path = attrs["region"]["path"]
+    column_name = attrs["instance_key"]
+    print(label_relative_path)
+    print(column_name)
+
+    # Check that ROI_table_obs has the right column and extract label_value
+    if column_name not in ROI_table_obs.columns:
+        raise ValueError(
+            'In preprocess_cellpose_input, "{column_name}" '
+            f" missing in {ROI_table_obs.columns=}"
+        )
+    label_value = int(ROI_table_obs[column_name][ROI_index])
+
+    # Load primary/current label arrays (lazily), and check shapes
+    primary_label_path = str(
+        Path(ROI_table_path).parent / label_relative_path / "0"
+    )
+    logger.critical(f"{primary_label_path=}")
+    primary_label_array = da.from_zarr(primary_label_path)
+    current_label_array = da.from_zarr(current_label_path)
+    logger.info(
+        f"[preprocess_cellpose_input] {primary_label_path=}, "
+        f"{primary_label_array.shape=}"
+    )
+    logger.info(
+        f"[preprocess_cellpose_input] {current_label_path=}, "
+        f"{current_label_array.shape=}"
+    )
+
+    # Load ROI data for current label array
+    current_label_region = current_label_array[region].compute()
+
+    # Load ROI data for primary label array, with or without upscaling
+    if primary_label_array.shape != current_label_array.shape:
+        logger.info("Upscaling of primary label is needed")
+        lowres_region = convert_region_to_low_res(
+            highres_region=region,
+            highres_shape=current_label_array.shape,
+            lowres_shape=primary_label_array.shape,
+        )
+        primary_label_region = primary_label_array[lowres_region].compute()
+        primary_label_region = upscale_array(
+            array=primary_label_region,
+            target_shape=current_label_region.shape,
+        )
+    else:
+        primary_label_region = primary_label_array[region].compute()
+
+    # Check that all shapes match
+    shapes = (
+        primary_label_region.shape,
+        current_label_region.shape,
+        image_array.shape[1:],
+    )
+    if len(set(shapes)) > 1:
+        raise ValueError(
+            "Shape mismatch:\n"
+            f"{current_label_region.shape=}\n"
+            f"{primary_label_region.shape=}\n"
+            f"{image_array.shape=}"
+        )
+
+    # Compute background mask
+    background_3D = primary_label_region != label_value
```

Comment:
Let's try to avoid dimensionality assumptions within our processing function, especially when we want to make them more general.
I think this one should hold for all cellpose processing of labels, but e.g. if we applied it to intensity images, those can already be 4D when they are multi-channel

Reply:
Moving this discussion to #340.